### PR TITLE
Run `testthat` Tests

### PR DESCRIPTION
**Pull request by PhilBoileau**
This PR provides functions for running `testthat` tests stored in the `tests/testthat` directory of a `simChef` simulation study. This is accomplished through two new functions:

- `run_tests()`: This function searches for `tests/testthat.R`. If it finds this file, then it is ran. Otherwise, an error message stating that the file can't be found is printed to the console.
- `test_sim_dir()`: This function wraps around `testthat::test_dir()`, running the dgp-, method-, evaluator- and visualizer-related tests in their respective directories. If using `create_sim()`, this function is automatically added to `tests/testthat.R`.

These functions are somewhat brittle. They work best with simulation studies created with `create_sim()`. If they are adopted, we'll need to modify documentation, like the "Setting Up Your Simulation" vignette, such that users are advised to generate new simulation projects with `create_sim()`.

### CODE REVIEW

**Comment by PhilBoileau** [comment]
We need to include a function that automatically loads all the functions defined in the `R/` subdirectories, similar to `pkgload::load_all()`. It would be called in `tests/testthat.R`, though it might be helpful in other situations too.

Something like this should work:

```{R}
load_all_sim_functions <- function() {
  sim_functions_files = list.files(
    c("R/dgp", "R/method", "R/eval", "R/viz"),
    pattern = "*.R$", full.names = TRUE, ignore.case = TRUE
  )
  sapply(sim_functions_files, source)
}
```

**Comment by jpdunc23** [comment]
@PhilBoileau This is awesome! Hopefully we can get it merged in before the JOSS submission.

A few of TODOs:

- [ ] Add any missing package dependencies to `DESCRIPTION` in the `Imports` field; the one I noticed from the failing tests is `testthat` needs to be moved from `Suggests` to `Imports`, which may be all that's missing.
- [ ] Remove `@importFrom` from Roxygen comments. Instead, using `package::function` syntax (which I see you already are) is sufficient.
- [ ] I think using internal function from other packages via `:::` is a no-go for `R CMD check` (see [here](https://r-pkgs.org/R-CMD-check.html#r-code)), so we need to replace those functions with some alternative, either from a public API or by copying the other package's internal code to `simChef`. I'd prefer to avoid the latter, but it's fine if need be.

Let me know if you have any questions or want some help with these!

**Comment by PhilBoileau** [comment]
Sounds good, thanks for these pointers! I'll try to knock these out by early next week.

On the last TODO: I think I had left that there so that we could discuss the best path forward. We could try reaching out to package maintainers to see if they'd consider exporting the functions we need. In the meantime, we can copy these internal functions into `simChef`. How does that sound?

**Comment by tiffanymtang** [comment]
Thanks so much, Phil! What are your thoughts on using `usethis:: use_git()` instead of `use_git_no_ask()`? I think this would avoid all instances of `:::`. The issue is I'm not familiar enough with `usethis::use_git()` nor `use_git_no_ask()` to understand if the added benefit of `use_git_no_ask()` is worth figuring out how to workaround the `:::`.

**Comment by PhilBoileau** [comment]
Just finished making the requested changes, @jpdunc23 and @tiffanymtang. I had replaced `usethiss::use_git()` by `use_git_no_ask()` because the former asks the user whether they want to commit the newly created files. Past-Phil must have wanted these files to be commited automatically. I don't think it's an issue though.

**Comment by PhilBoileau** [comment]
I'll update the "Setting Up Your Simulation Study" vignette to reflect these additions. I'll ping you both once that's done for a review.

File: vignettes/simulation-directory-setup.Rmd [hunk 1]
```diff
@@ -17,30 +17,28 @@ knitr::opts_chunk$set(
...
-If you're unfamiliar with git or GitHub, or haven't yet set git up on you
-machine, we suggest reviewing Chapters 4 through 8 of [*Happy Git and GitHub for
-the useR*](https://happygitwithr.com/index.html).
+the `create_sim()` function. Briefly, this function takes as argument a `path`
+indicating the location of the new directory for your simulation study and
+creates the necessary folders and files. It also initializes a new project and a
+git repository. This repository can optionally be linked to GitHub using, for
+example,
+[`usethis::use_github()`](https://usethis.r-lib.org/reference/use_github.html).
+If you don't want to use git, set `create_sim()`'s `init_git` argument can to
```
Comment (tiffanymtang) [inline vignettes/simulation-directory-setup.Rmd:37]: Typo: remove "can"

**Pull request merged**